src/ResourceManager/Common/Commands.ResourceManager.Common/Tags/TagsConversionHelper.cs

```diff
@@ -103,6 +103,11 @@ public static Hashtable CreateTagHashtable(IDictionary<string, string> dictionar
                 Hashtable tags = tagsFromCli as Hashtable;
                 return TagsConversionHelper.CreateTagDictionary(tags, validate: true);
             }
+            else if (cmdlet.MyInvocation.BoundParameters.TryGetValue("Tag", out tagsFromCli))
```

Comment:
nit: seems like this can be combined with above check. i.e.
`if (cmdlet.MyInvocation.BoundParameters.TryGetValue("Tags", out tagsFromCli) || cmdlet.MyInvocation.BoundParameters.TryGetValue("Tag", out tagsFromCli))`

for bonus points, add helper function :)

`if (TryGetValue("Tags", out tagsFromCli) || TryGetValue("Tag", out tagsFromCli))`